### ISSUE TITLE
The entering of digits in the Windows Calculator is not announced by NVDA if "Speak typed characters" is set to "Only in edit controls"

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#17670]
### Steps to reproduce:
1. Install and launch the latest NVDA alpha build.
1. Set NVDA's option "Speak typed characters" to "Only in edit controls".
1. Launch the Windows Calculator app.
1. Start entering digits.

### Actual behavior:
Nothing is announced.

### Expected behavior:
The entered digits are announced while being entered.

### System configuration
- Operating system: Windows 11 Pro version 24H2 (build 26100.2894), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-35158,aa828989 (2025.1.0.35158), installed, in bulgarian.
- Windows Calculator version: 11.2411.1.0, in bulgarian.

#### Does the issue still occur after restarting your computer?
Yes, it does.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes - 2024.4.2 - it behaves as expected.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes, it is.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
In my opinion, this question is not applicable / not relevant, because the reason is in an intentional change in NVDA's behavior.

### Additional information
In my opinion, the unexpected behavior (the bug) is due to the change in how NVDA announces typed characters. In version 2024.4.2 and earlier of NVDA, the typed characters would be announced always, no matter if they are entered in an edit field or not. With some recent changes in the alpha versions of NVDA, by default the typed characters are announced only if entered in an edit field. If the "Speak typed characters" option is set to "Always", the entered digits in the Calculator app are announced as expected. So, my proposal is to modify the NVDA's app module for the Calculator app, so in it the entered characters are announced even if the "Speak typed characters" option is set to "Only in edit controls".

**Comment by zstanecic** [comment]
Hi Kostadin,
This is normal and intentional behaviour. 
The windows calculator app doesn't have edit controls. 
So, if you set the option to speak character for edit controls only, it will do so for edit controls only. It is not a bug, it is a feature.
If you set it to speak always, it will use the old behaviour from the previous NVDA versions.

**Comment by ruifontes** [comment]
Should not Calculator be treated as an exception?
For me the problem is solved with a profil for calculator...
unfortunately, not all users know profiles or how to use them...

**Comment by Adriani90** [comment]
cc: @cary-rowen

**Comment by josephsl** [comment]
Hi,

Technical: the "display is ..." field in Calculator is not really an edit control as it shows expressions and results. Also, this field is updated when entering digits and other commands while not focused on this field (this point is important). While config profiles would be a viable solution, I think Rui is right in that an exception should be granted for cases like modern Calculator.

If @cary-rowen can access modern Calculator in Windows 10 and 11, then it might be possible to get it resolved using their knowledge. If not, as the author of modern Calculator app module, I'll do something about it unless someone wants to take this up.

Thanks.

**Comment by cary-rowen** [comment]
Yes, this should definitely be regarded as the exception. I am not sure whether the recent time allows me to implement this, and welcomes others to provide PR for it.

**Comment by gerald-hartig** [comment]
We support setting the default of "speak typed characters" back to "always".

**Comment by cary-rowen** [comment]
I think changing the default value to "Always" cannot solve this problem, or similar problems.

Undoubtedly, people who use new behaviors still want to hear the typed numbers in this case.

Therefore, whether the default behavior is "always" is not critical.

**Comment by CyrilleB79** [comment]
@gerald-hartig I second what @cary-rowen writes:
You may consider that speak typed characters "Only in edit controls" is not mature enough, and thus force "Always" by default. That's fine.

But this does not solve this issue, i.e. the behavior in Calculator when speak typed characters is "Only in edit controls" remains unexpected. 

If possible and not too difficult, I'd really recommend to fix the calculator's use case in 2025.1, even if the new value for speak typed characters is not the default, so that people who try this new parameter can have a good first opinion of this feature.

**Comment by SaschaCowley** [comment]
The issue is not the maturity of the "only in edit controls" feature, but that calculator is not an edit control. The feature is working as expected.

**Comment by CyrilleB79** [comment]
@SaschaCowley true, the calculator is not an edit control.

But we should think to the use case why"only in edit control" is selected. When this value is selected, the user expects to have typing reported when typing whenever text is really typed somewhere, even if this somewhere is not really an edit field.

For example, "only in edit fields" report char/word typing in terminals (e.g. cmd), even if terminal is not an edit field.

IMO, our appModule for calculator is trying to mimick an edit field behaviour, even if there isn't; e.g. see #13383. Today (2024.4.2), when you type a digits, NVDA discards the UIA notification "Display is xyz" for any single digit so that we can let NVDA's normal typing reporting operate.

You may disagree with me but telling that there is no edit field in Calculator is not enough.

The real question to answer is: Do we we want to mimick in some way an edit field in Calculator? Or do we just accept to hear what notifications report to us, i.e. "Display is xyz" each time we type something?

**Comment by cary-rowen** [comment]
cc @seanbudd 
I will fix this issue within 2025.1. So can it be added to the 2025.1 milestone?

**Comment by seanbudd** [comment]
No, we are no longer accepting general changes to 2025.1
